### DISCUSSION TITLE
Can we Whitelist InterviewBit Compiler's?

### DISCUSSION THREAD

**Discussion by letscode-17** [forem/forem#16301] [Completed]
Hey Team,

Can we get InterviewBit Compiler's whitelisted on dev.to? 
[https://www.interviewbit.com/online-c-compiler/](https://www.interviewbit.com/online-c-compiler/)]

As I see a lot of similar platforms like repl.it listed.

**Comment by rishabhrishabh** [comment]
That's pretty good idea !! We should do it.

**Comment by ellativity** [comment]
Thanks for this feature request @letscode-17 - can we ask what you mean by "whitelisted on dev.to", please? Do you have screenshot examples of how you'd like to use InterviewBit on DEV?

**Reply by letscode-17** [reply]
Hey @ellativity , so basically just to have Interviewbit compilers embedded to any post. As of now, it ain't happening or is there a way to get it embedded? 
For reference: https://dev.to/devteam/ways-to-enhance-your-dev-posts-useful-embeds-1g0f

**Reply by ellativity** [reply]
Thanks for explaining!

We're currently in the middle of a revision of the way embeds work on DEV and other Forems, so we're not adding any new liquid tags to the existing list. We hope to announce the new process soon and will bear your request in mind. 

Please sit tight while we get this completed!

**Reply by letscode-17** [reply]
Hey @ellativity, I just wanted to follow up on this request.

**Reply by ellativity** [reply]
Hey @letscode-17 you don't need to follow up on these requests. We'll reach back out to you if we need more info if/once someone gets to work on this. Thanks!

**Reply by letscode-17** [reply]
Hey @ellativity Thanks for your response. Is there a way we can get to know if this feature request would be considered(accepted/rejected) by the team.

**Reply by jeremyf** [reply]
@letscode-17 we're working on rolling out embeds for any page that has Open Graph and Twitter Card content.  Would that resolve this issue without a need for custom code?

**Reply by letscode-17** [reply]
Hey @jeremyf could you please elaborate on the new process of enabling embeds. IMO, it wouldn't support. 

Highlighting the steps of embedding IB snippets for your reference onto any post:

1. Once we run and save a program: [Link](https://www.interviewbit.com/online-python-compiler/), snippets get generated. eg:  [Snippet](https://www.interviewbit.com/snippet/5499e7b9038e98905382/)
2. On saving this snippet, you would see a pin like option (on hovering that button,  text reads as: Embed a Snippet.)
3. To embed the above snippet, the iframe tag that gets generated needs to be used. 

Please do let me know if this helps. 
Also, may I know how do you support the repl.it.com compilers on dev.to, could ours be supported in a similar fashion.

**Reply by jeremyf** [reply]
We're focusing on leveraging OpenGraph and Twitter Card meta content for general embeds.  If that meta data is in your snippet page, we'll show that rich content.  And that's where we're placing most of our effort.

`{% embed https://www.inteerviewbit.com/snippet/5499e7b9038e98905382/ %}`

So, if that is inadequate, you could consider submitting a pull request.  The [Replit Tag](https://github.com/forem/forem/blob/main/app/liquid_tags/replit_tag.rb) provides an example.  As someone responsible for maintaining these tags, I'm very conscious of both maintenance burden as well as any privacy considerations.

**Reply by jeremyf** [reply]
One caveat is that we did put a moratorium on new Embed tags, as they create an ever growing surface area.  Our latest updates (leveraging Twitter Cards and Open Graph) are our method of trying to conform to a web standard / ubiquitous interface.

**Comment by jeremyf** [comment]


**Reply by ellativity** [reply]
Jinx!

**Reply by jeremyf** [reply]
Gah, tried to delete my duplicate question.

**Comment by JennieOcken** [comment]
Hi @letscode-17 

The new embed functionality that supports any Open Graph or Twitter structure has been rolled out.  I believe this should solve your issue in a way that is cleanest for our code base.  Is it working for you?

You can now embed any content by using the `{% embed URL %}` structure and it will pick up Open Graph or Twitter Card on that page.
